python/ray/util/sgd/v2/session.py

```diff
@@ -128,6 +135,47 @@ def shutdown_session():
     _session = None
 
 
+def get_dataset_shard() -> Optional[RayDataset]:
```

Comment:
We'll need to support multiple shards for cross-validation, etc -- any way to rename to better reflect that?